server/pages/index.js

```diff
@@ -375,20 +378,44 @@ module.exports = function() {
 		app.get( '/themes/:theme_slug', function( req, res ) {
 			const context = getDefaultContext( req );
 
-			if ( config.isEnabled( 'server-side-rendering' ) ) {
+			const renderThemeSheet = function( theme ) {
 				const store = createReduxStore();
+				store.dispatch( {
+					type: ActionTypes.RECEIVE_THEME_DETAILS,
+					themeId: theme.id,
+					themeName: theme.name,
+					themeAuthor: theme.author,
+					themeScreenshot: theme.screenshot,
+				} );
```

Comment:
Could this call go somewhere deeper in the component hierarchy, e.g. inside `ThemesListFetcher` (to provide some initial state there, also on the client side)? Thinking about how to keep this part of `server/pages` as generic as possible for isomorphic routing.